SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_UnownFinder.cpp

```diff
@@ -0,0 +1,288 @@
+            return false;
+        }
+        if (start_region == MapRegion::NONE){
+            start_region = current_region;
+        }else if (start_region == current_region){
+            break;
+        }
+
+        if (current_region == region)
+        {
+           break;
+        }
+        pbf_press_dpad(env.console, DPAD_RIGHT, 20, 40);
+        env.console.botbase().wait_for_all_requests();
+    }
+
+    //Enter region
+    mash_A_to_change_region(env, env.console);
+    return true;
+}
+
+bool UnownFinder::pick_mount(SingleSwitchProgramEnvironment &env, MountState &mount_on, MountState &mount_off){
+    bool error = true;
+    MountDetector mount_detector;
+    for (size_t c = 0; c < 10; c++){
+        MountState mount = mount_detector.detect(env.console.video().snapshot());
+        if (mount == mount_off){
+            pbf_press_button(env.console, BUTTON_PLUS, 20, 105);
+            error = false;
+            break;
+        }
+        if (mount == mount_on){
+            pbf_wait(env.console, 5 * TICKS_PER_SECOND);
+            error = false;
+            break;
+        }
+        pbf_press_dpad(env.console, DPAD_LEFT, 20, 50);
+        env.console.botbase().wait_for_all_requests();
+    }
+    if (error){
+        throw OperationFailedException(env.console, "Unable to find Braviary after 10 attempts.");
+    }
+
+    return true;
+}
+
+void ruins_entrance_route(const BotBaseContext& context){
+    pbf_wait(context, (uint16_t)(0.6 * TICKS_PER_SECOND));
+    pbf_move_left_joystick(context, 139, 120, 10, 10);
+    pbf_wait(context, (uint16_t)(1.3 * TICKS_PER_SECOND));
+
+    pbf_press_button(context, BUTTON_B, (uint16_t)(9.5 * TICKS_PER_SECOND), 10);
+    pbf_wait(context, (uint16_t)(0.8 * TICKS_PER_SECOND));
+    pbf_move_left_joystick(context, 110, 90, 20, 10);
+
+    pbf_press_dpad(context, DPAD_LEFT, 10, 10);
+    pbf_press_button(context, BUTTON_PLUS, 10, 10);
+}
+
+void enter_ruins(const BotBaseContext& context){
+    pbf_press_button(context, BUTTON_B, (uint16_t)(4.5 * TICKS_PER_SECOND), 10);
+    pbf_wait(context, (uint16_t)(1.5 * TICKS_PER_SECOND));
+    pbf_move_left_joystick(context, 128, 255, 10, 0);
+    pbf_press_button(context, BUTTON_B, (uint16_t)(2 * TICKS_PER_SECOND), 10);
+}
+
+
+bool UnownFinder::run_iteration(SingleSwitchProgramEnvironment& env){
+    Stats& stats = env.stats<Stats>();
+
+    stats.attempts++;
+
+    enter_map(env);
+
+    MapRegion target_region = MapRegion::MIRELANDS;
+    if (!select_map(env, target_region)) return false;
+
+    MountState target_mount_on = MountState::BRAVIARY_ON;
+    MountState target_mount_off = MountState::BRAVIARY_OFF;
+    if (!pick_mount(env, target_mount_on, target_mount_off)) return false;
+
+    //Start path
+    env.console.log("Beginning Shiny Detection...");
+    {
+        ShinySoundDetector shiny_detector(env.console, SHINY_DETECTED.stop_on_shiny());
+
+        if(SKIP_PATH_SHINY){
```

Comment:
If you transpose this function a bit, you can have `run_until()` run while ignoring shinies if you set the flag to false. So even though it's ignoring shinies, it'll still report them and add them to the stats.

So what you could do here is to have two separate `run_until()` sessions. The first is the route to the ruins. That one you can set the stop-on-detected flag according to whether the user wants to ignore shinies. Once you reach the ruins, you can launch a 2nd `run_until()` with shiny stop enabled to run into the ruins.

Reply:
Hi, I'm changing to follow this approach but I'm having some issues on the `on_shiny_sound` as it expects a `ShinyDetectedActionOption`. 
I'm trying to create a new one but I can't seem to find a way to set the action to `IGNORE`

This is how it looks:

```
        ShinyDetectedActionOption SHINY_DETECTED_ON_ROUTE("0");
        ShinySoundDetector shiny_detector_route(env.console, !SKIP_PATH_SHINY );
        run_until(
            env, env.console,
            [](const BotBaseContext& context){
               ruins_entrance_route(context);
            },
            { &shiny_detector_route }
        );

        if (shiny_detector_route.detected()){
           stats.shinies++;
           on_shiny_sound(env, env.console, SHINY_DETECTED_ON_ROUTE, shiny_detector_route.results());
        }
```
But because the new `SHINY_DETECTED_ON_ROUTE` have no ACTION set, it also stops when hearing a shiny. Only way I could think was to convert the `SKIP_PATH_SHINY` boolean to a `ShinyDetectedActionOption` and let the user set like for the other one.
Any suggestions?